### ISSUE TITLE
Typo in docks deepMerge()

### ISSUE THREAD

**Issue by iz9** [facebook/immutable-js#1562]
https://facebook.github.io/immutable-js/docs/#/mergeDeep
import {merge} . Suppose to be import {deepMerge}
importing merge leads to confusing behavior - no expected result // { x: { y: 123, z: 456 }}.

**Comment by bohdanbirdie** [comment]
Seems to be fixed 
https://github.com/facebook/immutable-js/blob/master/type-definitions/Immutable.d.ts#L5179

**Comment by iz9** [comment]
You are right, there is correct import in d.ts file. But in code example in documentation page (url: https://facebook.github.io/immutable-js/docs/#/mergeDeep )  no changes yet ( const { merge } = require('immutable@4.0.0-rc.9') )
